### ISSUE TITLE
Invalid unused export warning on  1.75.0-nightly (cd674d617 2023-10-24)

### ISSUE THREAD

**Issue by dignifiedquire** [rust-lang/rust#117120]
CI started failing for us, without code changes in this area today with nightly

The code is sth like this

```rust
mod keys;
pub use keys::*;
```

and produces 

```
error: unused import: `keys::*`
  --> iroh-sync/src/lib.rs:44:9
   |
44 | pub use keys::*;
   |         ^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`
```

but it should not, given that this is a public export in `src/lib.rs`.

### Version with regression

```
latest update on 2023-10-24, rust version 1.75.0-nightly (cd674d617 2023-10-24)
```

### Logs

CI logs: https://github.com/n0-computer/iroh/actions/runs/6624632107/job/17994903346?pr=1708

**Comment by dignifiedquire** [comment]
Okay, I am very confused, now this got triggered on stable as well: https://github.com/n0-computer/iroh/actions/runs/6625039736/job/17995549252

```
stable-aarch64-apple-darwin unchanged - rustc 1.73.0 (cc66ad468 2023-10-03)
```

**Comment by dignifiedquire** [comment]
Can reproduce the warning on my local machine using

```
cargo +nightly --version
cargo 1.75.0-nightly (d2f6a0485 2023-10-20)
```

but 
```
cargo --version
cargo 1.73.0 (9c4383fb5 2023-08-26)
```
does not trigger it

**Comment by bvanjoi** [comment]
This behavior introduced through https://github.com/rust-lang/rust/pull/116033.

And I think this behavior is indeed expected because ﻿`iroh-sync::keys::*` has been re-exported in `﻿iroh-sync::sync` so `pub use keys::*` is not used: https://github.com/n0-computer/iroh/blob/main/iroh-sync/src/sync.rs#L28

**Comment by dignifiedquire** [comment]
I see, thanks a lot, that was definitely on the confusing side, but glad the compiler catches it now :)
